contracts/interchain-token-service/src/contract.rs

```diff
@@ -140,53 +142,51 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
         token_meta_data: TokenMetadata,
         initial_supply: i128,
         minter: Option<Address>,
-    ) -> Result<Address, ContractError> {
+    ) -> Result<(Address, BytesN<32>), ContractError> {
         caller.require_auth();
 
-        let mut initial_minter: Option<Address> = None;
-
-        if initial_supply > 0 {
-            initial_minter = Some(env.current_contract_address());
-        } else if let Some(ref minter_addr) = minter {
+        let initial_minter = if initial_supply > 0 {
+            Some(env.current_contract_address())
+        } else if let Some(ref minter) = minter {
             ensure!(
-                *minter_addr != env.current_contract_address(),
+                *minter != env.current_contract_address(),
                 ContractError::InvalidMinter
             );
-            initial_minter = minter.clone();
-        }
+            Some(minter.clone())
+        } else {
+            None
+        };
 
-        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt.clone());
-        let token_id = Self::interchain_token_id(env, caller.clone(), salt);
+        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
+        let token_id = Self::interchain_token_id(env, None, deploy_salt);
```

Comment:
```suggestion
        let token_id = Self::interchain_token_id(env, AddressZero, deploy_salt);
```
use a zero address placeholder similar to ITS factory. There's a reason we don't omit the param from the derivation itself, since it can cause collision issues. We'll revisit this logic afterwards, but for now, it should match EVM

Reply:
updated to use `GAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWHF`. reference: https://github.com/stellar/js-stellar-base/blob/master/test/unit/address_test.js